### ISSUE TITLE
Azure Bicep deployment fails after last release of virtual environments

### ISSUE THREAD

**Issue by SeidChr** [actions/virtual-environments#4080]
### Description

Using the powershell az module and a New-AzResourceGroupDeployment in an azure-devops pipeline started failing after the last release of these Virtual Environments.

Last successful build around noon Sept. 9th (CEST)
First failed build around noon Sept. 10th (CEST)

I have tried upgrading the module and bicep within the same build step while using different environments, but wasn't able to achieve any change in behavior.

The error itself resembles more or less exactly this behavior:
https://github.com/Azure/bicep/discussions/3140
`New-AzResourceGroupDeployment : Unexpected character encountered while parsing value: @. Path '', line 0, position 0.`

Can someone help us out with this?
Currently there is no workaround for us. Help is highly appreciated. Already spend most of my yesterday with this problem .

### Virtual environments affected

- [ ] Ubuntu 16.04
- [ ] Ubuntu 18.04
- [ ] Ubuntu 20.04
- [ ] macOS 10.15
- [ ] macOS 11
- [X] Windows Server 2016
- [X] Windows Server 2019
- [X] Windows Server 2022

### Image version and build link

2021-09-13T15:15:55.6759692Z ##[group]Virtual Environment
2021-09-13T15:15:55.6759917Z Environment: windows-2019
2021-09-13T15:15:55.6760140Z Version: 20210907.4
2021-09-13T15:15:55.6760478Z Included Software: https://github.com/actions/virtual-environments/blob/win19/20210907.4/images/win/Windows2019-Readme.md
2021-09-13T15:15:55.6760937Z Image Release: https://github.com/actions/virtual-environments/releases/tag/win19%2F20210907.4
2021-09-13T15:15:55.6761228Z ##[endgroup]

### Is it regression?

https://github.com/actions/virtual-environments/releases/tag/win16%2F20210901.3

### Expected behavior

Bicep deplyoment works as expected

### Actual behavior

https://github.com/Azure/bicep/discussions/3140
`New-AzResourceGroupDeployment : Unexpected character encountered while parsing value: @. Path '', line 0, position 0.`

FULL (redacted) LOG:
```
2021-09-13T15:25:05.5306482Z ##[section]Starting: Deploy Bicep Resources
2021-09-13T15:25:06.1950997Z ==============================================================================
2021-09-13T15:25:06.1951670Z Task         : Azure PowerShell
2021-09-13T15:25:06.1952381Z Description  : Run a PowerShell script within an Azure environment
2021-09-13T15:25:06.1952952Z Version      : 5.185.0
2021-09-13T15:25:06.1953419Z Author       : Microsoft Corporation
2021-09-13T15:25:06.1953986Z Help         : https://aka.ms/azurepowershelltroubleshooting
2021-09-13T15:25:06.1954667Z ==============================================================================
2021-09-13T15:25:07.1915062Z Generating script.
2021-09-13T15:25:07.1993029Z ========================== Starting Command Output ===========================
2021-09-13T15:25:07.2277238Z ##[command]"C:\Program Files\PowerShell\7\pwsh.exe" -NoLogo -NoProfile -NonInteractive -ExecutionPolicy Unrestricted -Command ". 'D:\a\_temp\9b1610b9-9462-4b54-896c-f059dcf292e5.ps1'"
2021-09-13T15:25:09.4939090Z Added TLS 1.2 in session.
2021-09-13T15:25:10.0396879Z ##[command]Import-Module -Name C:\Modules\az_6.1.0\Az.Accounts\2.5.3\Az.Accounts.psd1 -Global
2021-09-13T15:25:11.9148976Z ##[command]Clear-AzContext -Scope CurrentUser -Force -ErrorAction SilentlyContinue
2021-09-13T15:25:13.1480882Z ##[command]Clear-AzContext -Scope Process
2021-09-13T15:25:13.7090667Z ##[command]Connect-AzAccount -ServicePrincipal -Tenant *** -Credential System.Management.Automation.PSCredential -Environment AzureCloud @processScope
2021-09-13T15:25:14.6669590Z ##[command] Set-AzContext -SubscriptionId ############REDACTED############ -TenantId ***
2021-09-13T15:25:15.3949979Z DEBUG: Run in Pipeline:     True
2021-09-13T15:25:15.3969263Z DEBUG: Working Directory:   D:\a\r1\a
2021-09-13T15:25:15.3974240Z DEBUG: Resource Group Name: ############REDACTED############
2021-09-13T15:25:15.3987417Z DEBUG: Bicep File:          ############REDACTED############.bicep
2021-09-13T15:25:15.3988242Z DEBUG: Bicep File Exists:   True
2021-09-13T15:25:15.3988739Z Starting Azure-Deployment ...
2021-09-13T15:25:15.4130390Z 
2021-09-13T15:25:15.4146665Z CommandType     Name                                               Version    Source
2021-09-13T15:25:15.4147626Z -----------     ----                                               -------    ------
2021-09-13T15:25:15.4148406Z Cmdlet          New-AzResourceGroupDeployment                      4.2.0      Az.Resources
2021-09-13T15:25:15.4378806Z 
2021-09-13T15:25:15.4396142Z Name              : Az.Resources
2021-09-13T15:25:15.4416525Z Path              : C:\Modules\az_6.1.0\Az.Resources\4.2.0\Az.Resources.psm1
2021-09-13T15:25:15.4417496Z Description       : Microsoft Azure PowerShell - Azure Resource Manager and Active Directory cmdlets in Windows 
2021-09-13T15:25:15.4418719Z                     PowerShell and PowerShell Core.  Manages subscriptions, tenants, resource groups, deployment 
2021-09-13T15:25:15.4424432Z                     templates, providers, and resource permissions in Azure Resource Manager.  Provides cmdlets for 
2021-09-13T15:25:15.4425124Z                     managing resources generically across resource providers.
2021-09-13T15:25:15.4426121Z                     
2021-09-13T15:25:15.4428153Z                     For more information on Resource Manager, please visit the following: 
2021-09-13T15:25:15.4593634Z                     https://docs.microsoft.com/azure/azure-resource-manager/
2021-09-13T15:25:16.1821407Z                     For more information on Active Directory, please visit the following: 
2021-09-13T15:25:16.1828688Z                     https://docs.microsoft.com/azure/active-directory/fundamentals/active-directory-whatis
2021-09-13T15:25:16.1830623Z Guid              : 48bb344d-4c24-441e-8ea0-589947784700
2021-09-13T15:25:16.1831181Z Version           : 4.2.0
2021-09-13T15:25:16.1831718Z ModuleBase        : C:\Modules\az_6.1.0\Az.Resources\4.2.0
2021-09-13T15:25:16.1832229Z ModuleType        : Script
2021-09-13T15:25:16.1832695Z PrivateData       : {PSData}
2021-09-13T15:25:16.1834991Z AccessMode        : ReadWrite
2021-09-13T15:25:16.1835622Z ExportedAliases   : {[Get-AzADServicePrincipalCredential, Get-AzADServicePrincipalCredential], 
2021-09-13T15:25:16.1836355Z                     [Get-AzResourceProviderAction, Get-AzResourceProviderAction], [Get-AzSubscriptionDeployment, 
2021-09-13T15:25:16.1837063Z                     Get-AzSubscriptionDeployment], [Get-AzSubscriptionDeploymentOperation, 
2021-09-13T15:25:16.1837782Z                     Get-AzSubscriptionDeploymentOperation]…}
2021-09-13T15:25:16.1838511Z ExportedCmdlets   : {[Add-AzADGroupMember, Add-AzADGroupMember], [Export-AzResourceGroup, Export-AzResourceGroup], 
2021-09-13T15:25:16.1839359Z                     [Export-AzTemplateSpec, Export-AzTemplateSpec], [Get-AzADAppCredential, Get-AzADAppCredential]…}
2021-09-13T15:25:16.1840008Z ExportedFunctions : {}
2021-09-13T15:25:16.1840423Z ExportedVariables : {}
2021-09-13T15:25:16.1840982Z NestedModules     : {Microsoft.Azure.PowerShell.Cmdlets.Resources, Microsoft.Azure.PowerShell.Cmdlets.ResourceManager, 
2021-09-13T15:25:16.1841922Z                     Microsoft.Azure.PowerShell.Cmdlets.Tags}
2021-09-13T15:25:16.1842228Z 
2021-09-13T15:25:16.1842640Z C:\Modules\az_6.1.0\Az.Resources\4.2.0\Az.Resources.psm1
2021-09-13T15:25:16.1843277Z DEBUG: 3:25:15 PM - NewAzureResourceGroupDeploymentCmdlet begin processing with ParameterSet 'ByTemplateFileWithNoParameters'.
2021-09-13T15:25:16.1844520Z DEBUG: 3:25:15 PM - using account id '***'...
2021-09-13T15:25:16.1845597Z DEBUG: [Common.Authentication]: Authenticating using Account: '***', environment: 'AzureCloud', tenant: '***'
2021-09-13T15:25:16.1846828Z DEBUG: 3:25:15 PM - [ServicePrincipalAuthenticator] Calling ClientSecretCredential.GetTokenAsync - ApplicationId:'***', TenantId:'***', Scopes:'https://management.core.windows.net//.default', AuthorityHost:'https://login.microsoftonline.com/'
2021-09-13T15:25:16.1847850Z DEBUG: ClientSecretCredential.GetToken invoked. Scopes: [ https://management.core.windows.net//.default ] ParentRequestId: 
2021-09-13T15:25:16.1848702Z DEBUG: Request [############REDACTED############] POST https://login.microsoftonline.com/***/oauth2/v2.0/token
2021-09-13T15:25:16.1849291Z x-client-SKU:REDACTED
2021-09-13T15:25:16.1849763Z x-client-Ver:REDACTED
2021-09-13T15:25:16.1850176Z x-client-CPU:REDACTED
2021-09-13T15:25:16.1850578Z x-client-OS:REDACTED
2021-09-13T15:25:16.1850994Z x-client-current-telemetry:REDACTED
2021-09-13T15:25:16.1851458Z x-client-last-telemetry:REDACTED
2021-09-13T15:25:16.1851905Z x-ms-lib-capability:REDACTED
2021-09-13T15:25:16.1852339Z client-request-id:REDACTED
2021-09-13T15:25:16.1852769Z return-client-request-id:REDACTED
2021-09-13T15:25:16.1853196Z x-app-name:REDACTED
2021-09-13T15:25:16.1853590Z x-app-ver:REDACTED
2021-09-13T15:25:16.1854046Z x-ms-client-request-id:############REDACTED############
2021-09-13T15:25:16.1854542Z x-ms-return-client-request-id:true
2021-09-13T15:25:16.1855080Z User-Agent:azsdk-net-Identity/1.4.0,(.NET 5.0.9; Microsoft Windows 10.0.17763)
2021-09-13T15:25:16.1855617Z Content-Type:application/x-www-form-urlencoded
2021-09-13T15:25:16.1856245Z client assembly: Azure.Identity
2021-09-13T15:25:16.1856762Z DEBUG: Response [############REDACTED############] 200 OK (00.1s)
2021-09-13T15:25:16.1857262Z Cache-Control:no-store, no-cache
2021-09-13T15:25:16.1857657Z Pragma:no-cache
2021-09-13T15:25:16.1858075Z Strict-Transport-Security:REDACTED
2021-09-13T15:25:16.1858530Z X-Content-Type-Options:REDACTED
2021-09-13T15:25:16.1858936Z P3P:REDACTED
2021-09-13T15:25:16.1859320Z client-request-id:REDACTED
2021-09-13T15:25:16.1859860Z x-ms-request-id:############REDACTED############
2021-09-13T15:25:16.1860329Z x-ms-ests-server:REDACTED
2021-09-13T15:25:16.1860733Z x-ms-clitelem:REDACTED
2021-09-13T15:25:16.1861136Z Set-Cookie:REDACTED
2021-09-13T15:25:16.1861569Z Date:Mon, 13 Sep 2021 15:25:14 GMT
2021-09-13T15:25:16.1862025Z Content-Type:application/json; charset=utf-8
2021-09-13T15:25:16.1862954Z Expires:-1
2021-09-13T15:25:16.1863341Z Content-Length:1427
2021-09-13T15:25:16.1863562Z 
2021-09-13T15:25:16.1864266Z DEBUG: ClientSecretCredential.GetToken succeeded. Scopes: [ https://management.core.windows.net//.default ] ParentRequestId:  ExpiresOn: 2021-09-13T16:25:14.0000000+00:00
2021-09-13T15:25:16.1865309Z DEBUG: [Common.Authentication]: Received token with LoginType 'User', Tenant: '***', UserId: '***'
2021-09-13T15:25:16.1898203Z ##[error]Unexpected character encountered while parsing value: p. Path '', line 5, position 0.
2021-09-13T15:25:16.1910018Z ##[error]PowerShell exited with code '1'.
2021-09-13T15:25:16.3215522Z ##[section]Finishing: Deploy Bicep Resources
```

### Repro steps

e.g.: https://github.com/Azure/bicep/discussions/3140#discussioncomment-1304188

test.bicep
```
output test string = 'helloworld from empty'
```

Az Ps Script
```
New-AzResourceGroupDeployment -TemplateFile test.bicep  -ResourceGroupName foo -verbose
```

**Comment by al-cheb** [comment]
Hello, @SeidChr 
Could you set Az module to version 6.3.0/6.4.0 in the task?

**Comment by SeidChr** [comment]
Hi @al-cheb,

I am not aware that it is possible to just set a specific version of a module. need to check.
In the end its a default Azure Devops Task (Azure Powershell)

**Comment by SeidChr** [comment]
Just found the setting. it was set up to be 'latest installed version'
I've set it to '6.4.0' and it just works out of the box!

Thank you @al-cheb !
I did really miss-interpret that setting to be for the powershell version, instead of the az-module version. Learned sth. today!

Guess the 'latest installed version' should still use 6.4.0 when its actually available. 
But regarding this very ticket, the issue is solved.
